### PR TITLE
Inaira Odin Data

### PR DESCRIPTION
**Pull request by ANeaves**
Adds the CMake structure and the plugins for Inaira Odin Data. Plugin uses cppflow framework to run an ML model that can receive and analyse frame data

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -127,3 +128,7 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+
+# Cppflow Stuff
+data/frameProcessor/include/cppflow
```
Comment (timcnicholls) [inline .gitignore]: If you're excluding cppflow includes here, how does the repo express a dependency on it? Should it be done with a CMake package finder (like e.g. ZeroMQ) instead?

File: data/frameProcessor/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Add the src subdirectory for frameProcessor
+add_subdirectory(src)
+add_subdirectory(test)
```
Comment (timcnicholls) [inline data/frameProcessor/CMakeLists.txt]: Lots of files in repo missing trailing newlines (this is just an example)

File: data/config/inaira-fp.json [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        "plugin": {
+            "connect": {
+                "index":"hdf",
+                "connection":"learning"
+            }
+        }
+    },
+    {
+        "learning": {
+            "model_path":"/aeg_sw/work/projects/inaira/casting/models/casting_prob_model/",
```
Comment (timcnicholls) [inline data/config/inaira-fp.json]: Config exposing details of our systems - committed code should have example paths and be customised once installed.

**Review by timcnicholls — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (ANeaves) [reply .gitignore]: Likely yes, it should be done the same way we include external packages like ZeroMQ. It was placed into the include folder as a "temporary" method to get it working, but I should be able to move it out of there and find it using cmake instead

**Review by timcnicholls — APPROVED** [review]

**Pull request merged**